### PR TITLE
[core] OCTRL-886 correctly use mutex in environment/manager.go

### PR DESCRIPTION
**Pull request by justonedev1**

### CODE REVIEW

**Comment by justonedev1** [comment]
I created PR before testing, so don't merge yet please. Sorry for confusion.

**Comment by justonedev1** [comment]
this PR can be merged, I tested it in open stack

**Review by teo — APPROVED** [review]

**Pull request merged**
